documentation/add/09_Serializable_Interfaces.md

```diff
@@ -0,0 +1,132 @@
+            end
+        end
+    end
+ ```
+
+Before the properties are set within the serializable class constructor one by one, the *serializable* code sets the protected property **check_combo_arg_** to false, so the check for each interdependent property is not occurring. The code then sets the property **check_combo_arg_** to true and runs the check method: *check_combo_arg* after all interdependent properties have been set, throwing if **check_combo_arg** returns **false**. As the result, user can not change interdependent properties in a wrong way and needs to use constructor or serializable interface to set defined class with interdependent properties.
+
+There is other way of setting interdependent properties assuring their validity, namely defining the methods, which allow setting all interdependent properties at once. E.g. setting `s`, `e`, `npix` array may be performed by introducing property `se_npix` with validates and sets all three properties together. This method may be more efficient from point of view of serializing data so will be used where it is justified and convenient.
+
+## Standard serializable class constructor form.
+
+As a *serializable* class has public interface, fully described by the list of properties, returned by *saveableFields* method, it makes sense to define standard constructor, which use this interface to define the contents of the serializable class. The constructor would have the form:
+```Matlab
+    obj = class_name(positional_val1,positional_val2,positional_val3,... '-key1',val1,'-key2',val2,...);
+```
+where *positional_val1, positional_val2...* are the values, one would set up to the public properties of the *serializable* interface in the order, these properties names are returned by *saveableFields* method. The `keys` then would be the names of these properties, followed by the appropriate values.
+
+To facilitate creation of such constructors, the *serializable* class has a method:
+
+```Matlab
+ [obj,remains] = set_positional_and_key_val_arguments(obj,...
+                param_names_list,varargin);
+```
+
+where `param_names_list` is the list of the property names. Then, a standard constructor for a serializable object would have a form:
+
+```Matlab
+    function obj = an_serializable_obj(varargin)
+    	if nargin == 0
+            return;
+        end
+        prop_names = obj.saveableFields();
+        obj = set_positional_and_key_val_arguments(obj,...
+                param_names_list,varargin{:});
+    end
+```
+
+## Generic eq operator.
+
+As serializable classes have public interfaces with values fully defining the state of the object, to compare serializable objects is reasonable to define `eq` operator, which would work by comparing the values, retrieved from public properties provided by *saveableFields* method of the serializable interface.
+
+To avoid code duplication, such operator should be defined directly on *serializable* interface.
```

Comment:
Can you do this? The actual eq would depend on the class definition

Reply:
It could be overloaded on a class by class basis and invoke the serializable class definiton if useful.

Reply:
I can do that -- all serializable classes expose the interface for comparison so `eq` will use the interface. Nobody prevents you from overloading parental `eq` operator to modify comparison details but the standard `eq` is good for majority of cases.